python_modules/dagster/dagster/_core/definitions/asset_automation_evaluator.py

```diff
@@ -24,10 +26,25 @@
     from .auto_materialize_rule import AutoMaterializeRule, RuleEvaluationResults
 
 
+class AutomationConditionNodeSnapshot(NamedTuple):
+    """A serializable snapshot of a node in the AutomationCondition tree."""
+
+    class_name: str
+    description: str
+    child_hashes: Sequence[str]
+
+    @property
+    def hash(self) -> str:
```

Comment:
I suspect we'll want to be able to make minor aesthetic tweaks to descriptions with minimal consequences. With this hashing scheme, minor changes to descriptions would cause hashes to change - would that have consequences?

If so, having a hash attribute instead of child hashes could give us more flexibility to control when hashes change / stay the same?

Reply:
I like that idea

Reply:
@sryza I ended up updating this upstack (here: https://github.com/dagster-io/dagster/pull/18684/files) to avoid rebasing this change through a bunch of PRs. Changed the name to "unique_id" to make it clearer what the purpose of the field was